Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -98,23 +98,76 @@ int VertexLoaderX64::ReadVertex(OpArg data, u64 attribute, int format, int count
 
 	int elem_size = 1 << (format / 2);
 	int load_bytes = elem_size * count_in;
-	if (load_bytes > 8)
-		MOVDQU(coords, data);
-	else if (load_bytes > 4)
-		MOVQ_xmm(coords, data);
-	else
-		MOVD_xmm(coords, data);
-
-	PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
 
-	if (format != FORMAT_FLOAT)
+	if (cpu_info.bSSSE3)
 	{
-		// Sign extend
+		if (load_bytes > 8)
+			MOVDQU(coords, data);
+		else if (load_bytes > 4)
+			MOVQ_xmm(coords, data);
+		else
+			MOVD_xmm(coords, data);
+
+		PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+
+		// Sign-extend.
 		if (format == FORMAT_BYTE)
 			PSRAD(coords, 24);
 		if (format == FORMAT_SHORT)
 			PSRAD(coords, 16);
+	}
+	else
+	{
+		switch (format)
+		{
+		case FORMAT_UBYTE:
+		case FORMAT_BYTE:
+			MOVD_xmm(coords, data);
+			PXOR(XMM1, R(XMM1));
+			PUNPCKLBW(coords, R(XMM1));
+			if (count_in > 1)
+				PUNPCKLWD(coords, R(XMM1));
```

Comment:
Why is this only necessary in the count_in > 1 case?  Don't you still need to insert zeros in the count_in == 1 case?